### PR TITLE
Added object and array functions.

### LINKED ISSUE #267: Add `omitNil` (and friends)
This is a common pattern:

```js
_.omitBy(_.isNil, x)
```

It's like `F.compactObject`, but based on `_.isNil` instead of truthiness. `F.compactObject` should be based on truthiness because lodash's `_.compact` for arrays does the same, but we often want to remove fields that are "missing" (nil).

While we're at it, we should also add `F.omitBlank` which does `_.omitBy(F.isBlank)` and maybe also `omitEmpty`.

### PR DESCRIPTION
**Pull request by geosp**
Fixes #267

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Your PR has no browser errors. Great job!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: 2b731896cea0e47b3449341aacbdf38993711f88


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 45d55bb59b2ee72f0813e93446c63bafc1492adc
</p>

File: README.md [hunk 1]
```diff
@@ -210,6 +210,12 @@ Haskell's [groupBy](http://zvon.org/other/haskell/Outputlist/groupBy_f.html).
 
 **Note:** Intersperse can be used with JSX components! Specially with the `differentLast` iterator:
 
+### replaceElementBy
+`(fn(array_element), value, array) -> array` Replaces an element in an array with `value` based on the boolean result of a function `fn`.
+
+### replaceElement
```
Comment (daedalus28) [inline README.md]: these docs were added in the middle of `intersperse`'s docs

File: README.md [hunk 2]
```diff
@@ -210,6 +210,12 @@ Haskell's [groupBy](http://zvon.org/other/haskell/Outputlist/groupBy_f.html).
...
+`(a, b, array) -> array` Replaces all elements equal to `a` in an array with `b`.
```
Comment (daedalus28) [inline README.md]: for the docs, let's give `a` and `b` clearer names

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "futil-js",
-  "version": "1.56.0",
+  "version": "1.56.1",
```
Comment (daedalus28) [inline package.json]: this should be 1.57.0

File: README.md (hunk 2 above)
Comment (geosp) [reply README.md]: Done

File: package.json (hunk 3 above)
Comment (geosp) [reply package.json]: Done

File: README.md (hunk 1 above)
Comment (geosp) [reply README.md]: Fixed

**Pull request merged**